### PR TITLE
Cleanup some build warnings

### PR DESCRIPTION
**Pull request by valarnin**
Just some build warning cleanup.

There's a few other overall issues:

1. FFXIV plugin deprecation:
```
\OverlayPlugin\OverlayPlugin.Core\MemoryProcessors\FFXIVMemory.cs(75,28): warning CS0618: 'FFXIVRepository.GetCurrentFFXIVProcess()' is obsolete: 'Subscribe to the ProcessChanged event instead (See RegisterProcessChangedHandler())'
\OverlayPlugin\OverlayPlugin.Core\MemoryProcessors\FFXIVMemory.cs(63,13): warning CS0162: Unreachable code detected
\OverlayPlugin\OverlayPlugin.Core\EventSources\CactbotEventSource.cs(761,13): warning CS0618: 'Registry.RegisterOverlayPreset(IOverlayPreset)' is obsolete: 'Please call RegisterOverlayPreset2() on the Registry object instead.'
\OverlayPlugin\OverlayPlugin.Core\EventSources\CactbotEventSource.cs(773,13): warning CS0618: 'Registry.RegisterOverlayPreset(IOverlayPreset)' is obsolete: 'Please call RegisterOverlayPreset2() on the Registry object instead.'
\OverlayPlugin\OverlayPlugin.Core\OverlayZCorrector.cs(38,27): warning CS0618: 'FFXIVRepository.GetCurrentFFXIVProcess()' is obsolete: 'Subscribe to the ProcessChanged event instead (See RegisterProcessChangedHandler())'
```
2. Cactbot code pulled into OverlayPlugin:
```
\OverlayPlugin\OverlayPlugin.Core\PluginMain.cs(426,21): warning CS0219: The variable 'foundCactbot' is assigned but its value is never used
```
3. This unused event:
```
\OverlayPlugin\OverlayPlugin.Core\OverlayBase.cs(29,49): warning CS0067: The event 'OverlayBase<TConfig>.OnLog' is never used
```

Resolution for the above:

1. I'll make a separate PR for this
2. Not sure exactly how we should handle this. Remove the previous effort entirely?
3. I'm not sure if this event is actually unused, could use a second opinion on if it's safe to remove this (and also the event listener being added in `PluginMain.cs:351`)

### CODE REVIEW

File: DEPS.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 deps = {
     'ACT': {
-        'url': 'https://github.com/EQAditu/AdvancedCombatTracker/releases/download/3.4.7.268/ACTv3.zip',
+        'url': 'https://github.com/EQAditu/AdvancedCombatTracker/releases/download/3.4.9.271/ACTv3.zip',
         'dest': 'Thirdparty/ACT',
         'strip': 0,
         'hash': ['sha256', 'adf13a38d0938ce90f8e674f8365b227d933b91636ddf72b26c85702f6e3b808'],
```
Comment (quisquous) [inline DEPS.py:10]: Does this hash need to be updated as well, then?

**Comment by quisquous** [comment]
Re: above issues.

1) separate PR sounds great, thank you!
2) unclear on what to do here; I think probably postpone for a bit? I'm happy to have discussion with others on what to do about cactbot-in-OverlayPlugin.  I think there's a lot that's desirable about combining the two plugins and moving cactbot to "just" being an overlay.  There's still a few open questions here though?
3) unclear to me as well; it seems unused in OverlayPlugin, in that nothing ever invokes the event even if PluginMain.cs subscribes to it.  However, it's public in `IOverlay` so it's possible something other OverlayPlugin plugin (not cactbot, at least) is invoking it?

File: DEPS.py (hunk 1 above)
Comment (valarnin) [reply DEPS.py:10]: Yeah, I can update the hash, but it appears as though either the `tools/fetch_deps.py` script either isn't checking the hash or isn't warning or erroring on a bad hash.

As an aside, running that script from a bash terminal (which is my only python install right now) mangles the FFXIV SDK paths and I have no where near the level of python knowledge to actually fix the issue.

```bash
ls Thirdparty/FFXIV_ACT_Plugin 
 FFXIV_ACT_Plugin.dll         'SDK\FFXIV_ACT_Plugin.Common.dll'  'SDK\FFXIV_ACT_Plugin.LogFile.dll'  'SDK\FFXIV_ACT_Plugin.Network.dll'  'SDK\FFXIV_ACT_Plugin.Parse.dll'
 FFXIV_ACT_Plugin_Korean.dll  'SDK\FFXIV_ACT_Plugin.Config.dll'  'SDK\FFXIV_ACT_Plugin.Memory.dll'   'SDK\FFXIV_ACT_Plugin.Overlay.dll'  'SDK\FFXIV_ACT_Plugin.Resource.dll'
```

Comment (quisquous) [reply DEPS.py:10]: Oof.  If you have a chance, could you file issues for both of those and maybe somebody (me?) can remember to take a look?

Comment (valarnin) [reply DEPS.py:10]: Hmm, even more issues with that hashing logic. Apparently if I'm reading it right, it only hashes the first 16kb of the file. And with that context (only the first 16kb of the file) the hashes of the two versions are identical:

```bash
$ curl https://github.com/EQAditu/AdvancedCombatTracker/releases/download/3.4.9.271/ACTv3.zip -L -o ACTv3.zip --silent
$ head -c $((1024 * 16)) ACTv3.zip | sha256sum
adf13a38d0938ce90f8e674f8365b227d933b91636ddf72b26c85702f6e3b808  -
$ curl https://github.com/EQAditu/AdvancedCombatTracker/releases/download/3.4.7.268/ACTv3.zip -L -o ACTv3.zip --silent
$ head -c $((1024 * 16)) ACTv3.zip | sha256sum
adf13a38d0938ce90f8e674f8365b227d933b91636ddf72b26c85702f6e3b808  -
```

Reference:

https://github.com/OverlayPlugin/OverlayPlugin/blob/9a63992c20b54d09d0b3842af0de5c421e4766d8/tools/fetch_deps.py#L98-L110

Comment (quisquous) [reply DEPS.py:10]: Oh wow.  Ok, I guess it's ok then!

**Review by quisquous — APPROVED** [review]

**Pull request merged**